app/controllers/iiif_controller.rb

```diff
@@ -115,23 +114,41 @@ def image_info
 
     info['sizes'] = [{ width: 400, height: 400 }] unless current_image.maybe_downloadable?
 
+    services = []
     if anonymous_ability.cannot? :download, current_image
-      info['service'] = {
-        '@id' => iiif_auth_api_url,
-        'profile' => 'http://iiif.io/api/auth/1/login',
-        'label' => 'Stanford-affiliated? Login to view',
-        'service' => [
-          {
-            '@id' => iiif_token_api_url,
-            'profile' => 'http://iiif.io/api/auth/1/token'
-          },
-          {
-            '@id' => logout_url,
-            'profile' => 'http://iiif.io/api/auth/1/logout',
-            'label' => 'Logout'
-          }
-        ]
-      }
+      if current_image.stanford_restricted?
+        services << {
+          '@id' => iiif_auth_api_url,
+          'profile' => 'http://iiif.io/api/auth/1/login',
+          'label' => 'Stanford-affiliated? Login to view',
+          'service' => [
+            {
+              '@id' => iiif_token_api_url,
+              'profile' => 'http://iiif.io/api/auth/1/token'
+            },
+            {
+              '@id' => logout_url,
+              'profile' => 'http://iiif.io/api/auth/1/logout',
+              'label' => 'Logout'
+            }
+          ]
+        }
+      end
+
+      if current_image.restricted_by_location? && !current_image.location_rights(current_user.location).first
+        services << {
+          'profile' => 'http://iiif.io/api/auth/1/external',
+          'label' => 'External Authentication Required',
+          'failureHeader' => 'Restricted Material',
+          'failureDescription' => "This material is only viewable on-site at #{current_image.restricted_locations}."
```

Comment:
i noticed that the embed text specifically mentions "playing" media, so i got advice from @jvine on more generic text for the message.

fixed and pushed.